common:
  lid_topic: "/livox/lidar"
  imu_topic: "/livox/imu"
  img_en: 1
  lidar_en: 1
  ros_driver_bug_fix: false

extrin_calib:
  # lidar->IMU
  extrinsic_T: [ 0.04165, 0.02326, -0.0284 ]
  extrinsic_R: [1, 0, 0,
                0, 1, 0,
                0, 0, 1]

  # Multi-camera extrinsics: camera-to-LiDAR transformation
  cameras:
    - cam_id: 0
      img_topic: "/left_camera/image"
      Rcl: [-0.00200, -0.99975, -0.02211,
             -0.00366, 0.02212, -0.99975,
             0.99999, -0.00192, -0.00371]
      Pcl: [0.00260, 0.05057, -0.00587]
#      Rcl: [-0.0036250, -0.9998907, -0.0143360,
#            0.0075568,  0.0143083, -0.9998690,
#            0.9999649, -0.0037329,  0.0075041]
#      Pcl: [0.00549469, 0.0712101, 0.0322054]

time_offset:
  imu_time_offset: 0.0
  img_time_offset: 0.1
  exposure_time_init: 0.0

preprocess:
  point_filter_num: 1
  filter_size_surf: 0.2
  lidar_type: 1
  scan_line: 6
  blind: 0.5


vio:
  max_iterations: 5
  outlier_threshold: 100
  img_point_cov: 1000
  patch_size: 8
  patch_pyrimid_level: 4
  normal_en: true
  inverse_composition_en: false
  exposure_estimate_en: true
  inv_expo_cov: 0.1

imu:
  imu_en: true
  imu_int_frame: 30
  acc_cov: 0.5 # 0.2
  gyr_cov: 0.3 # 0.5
  b_acc_cov: 0.0001 # 0.1
  b_gyr_cov: 0.0001 # 0.1

lio:
  max_iterations: 5
  dept_err: 0.02
  beam_err: 0.05
  min_eigen_value: 0.0025
  voxel_size: 0.5
  max_layer: 2
  max_points_num: 100
  layer_init_num: [5, 5, 5, 5, 5]

uav:
  imu_rate_odom: false
  gravity_align_en: false

publish:
  dense_map_en: true
  pub_effect_point_en: false
  pub_body_cloud_en: false
  pub_plane_en: false
  pub_scan_num: 1
  blind_rgb_points: 0.0

evo:
  seq_name: "frame_pose"
  pose_output_en: true

pcd_save:
  pcd_save_en: false
  colmap_output_en: false
  filter_size_pcd: 0.2
  interval: -1

